compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1050,12 +1050,39 @@ void OMR::X86::TreeEvaluator::compareIntegersForOrder(
              firstChild->getReferenceCount() == 1)
             {
             TR::MemoryReference  *tempMR = generateX86MemoryReference(firstChild, cg);
-            TR::TreeEvaluator::compareGPMemoryToImmediate(node, tempMR, static_cast<int32_t>(constValue), cg);
+            if (compareSize == 1)
+               {
+               generateMemImmInstruction(TR::InstOpCode::CMP1MemImm1, node, tempMR, static_cast<int32_t>(constValue), cg);
+               }
+            else if (compareSize == 2)
+               {
+               //shouldn't use Imm2 instructions
```

Comment:
Can you explain this comment?

Reply:
I'm following an approach taken in similar circumstances in compareIntegersForEquality() where 2-byte immediates are avoided in favor of using the MemReg form instead, which as I understand is due to the fact that using 16-bit immediate operands can result in slower instruction decoding because of the need for an operand-size prefix in the encoding to specify the 2-byte size.